azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -879,12 +893,24 @@ private void prepareJobProperties(final ExecutableNode node) throws IOException
       props = jobSource;
     }
 
+    if (this.azkabanProps.getBoolean(
+        ConfigurationKeys.EXECUTOR_PROPS_RESOLVE_OVERRIDE_EXISTING_ENABLED, false)) {
+      // Flow override props are configured to also override existing job props
+      // =>
+      // 5. If there are any runtime flow overrides, we apply them now.
+      final Map<String, String> flowParam =
+          this.flow.getExecutionOptions().getFlowParameters();
+      if (flowParam != null && !flowParam.isEmpty()) {
+        props.putAll(flowParam);
+      }
+    }
```

Comment:
@ypadron-in wrote in https://github.com/azkaban/azkaban/pull/2597#discussion_r566636794:

> With this block of code I think we're mixing the ability to overwrite properties with the concept of flow and job level properties.
>
> -By providing runtime properties we are overwriting the value of already defined properties (or adding new ones).
-That should NOT interfere with the notion of flow and job level properties. I believe flow level properties are meant to define properties to be applied to the entire DAG and as a side effect offer a convenient way of defining job level properties in one place that will be applied to all jobs. However, job level properties take precedence over the same properties defined at flow level.
>
> I don't think it's ok to add this piece of code because we're saying that flow level overwrites take precedence over the existing hierarchy of flow/job level properties. Flow level overwrites should only take precedence over flow level definition properties.
>
> With the "job specific flow overwrites" feature added by ~~this~~ PR (https://github.com/azkaban/azkaban/pull/2597) we're adding the missing piece in the puzzle which is the ability to overwrite job level definition properties.

and

> I think we should revert ~~that~~ this PR. Do you understand the issue there?

I understand that what you have described is the old/existing behavior, and it also remains the default. As you can see, this behavior is controlled by azkaban server property, and is disabled by default.

But the ability to override also existing job properties globally is the whole point of this PR. It's just very handy in practice to be able to override globally instead of listing all jobs (or nodes) that need some override separately.

I have actually failed to see what the real-life use cases are for overriding "flow level props", but still not overriding if that property is present in some jobs. I understand that the default behavior cannot be changed without ensuring that users understand it though.

To continue on the use case, to me it would be more natural to not set a flow override but just set job-specific (or node-specific) overrides where needed and not set a global (flow) override at all 🤷‍♂️ But of course either way can work, more or less.